### PR TITLE
Fix detection of a tag build

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: templates/github/.github/workflows/scripts/before_install.sh.j2 [hunk 1]
```diff
@@ -4,6 +4,20 @@
...
+  BRANCH_BUILD=0
+else
+  BRANCH_BUILD=1
+  BRANCH="${GITHUB_REF##refs/heads/}"
+fi
+if [ "${GITHUB_REF##refs/tags/}" = "${GITHUB_REF}" ]
+then
+  TAG_BUILD=0
+else
+  TAG_BUILD=1
```
Comment (fao89) [inline templates/github/.github/workflows/scripts/before_install.sh.j2:18]: I like it

**Review by fao89 — APPROVED** [review]

File: templates/github/.github/workflows/scripts/before_install.sh.j2 [hunk 2]
```diff
@@ -4,6 +4,20 @@
 
 set -mveuo pipefail
 
+if [ "${GITHUB_REF##refs/heads/}" = "${GITHUB_REF}" ]
```
Comment (daviddavis) [inline templates/github/.github/workflows/scripts/before_install.sh.j2:7]: I'm curious about what this check is doing. I guess I would have imagined we'd check something like `"$GITHUB_REF" = refs/heads*`?

Comment (mdellweg) [reply templates/github/.github/workflows/scripts/before_install.sh.j2:7]: Well the `##` removes `refs/heads/` from the front if it is there, and in that case the strings stop to be equal. So yes.

Comment (daviddavis) [reply templates/github/.github/workflows/scripts/before_install.sh.j2:7]: Hmm it seems like such an indirect way of checking that, no?

Comment (mdellweg) [reply templates/github/.github/workflows/scripts/before_install.sh.j2:7]: It is compatible with the all bourn shell flavors, afaik.

**Review by daviddavis — APPROVED** [review]

**Pull request merged**